packages/backend-defaults/src/entrypoints/cache/types.ts

```diff
@@ -28,12 +28,86 @@ export type RedisCacheStoreOptions = {
   cluster?: RedisClusterOptions;
 };
 
+/**
+ * Configuration for a single Infinispan server.
+ * @public
+ */
+export interface InfinispanServerConfig {
+  host: string;
+  port: number;
+}
+
+/**
+ * SSL/TLS options for the Infinispan client.
+ * @public
+ */
+export interface InfinispanSslOptions {
+  enabled: boolean;
+  secureProtocol?: string | null;
+  caFile?: string | null;
+  clientCertificateFile?: string | null;
+  clientKeyFile?: string | null;
+  clientKeyPassword?: string | null;
+  sniHostname?: string | null;
+}
+
+/**
+ * Authentication options for the Infinispan client.
+ * @public
+ */
+export interface InfinispanAuthOptions {
+  enabled: boolean;
+  saslMechanism?: string | null;
+  userName?: string | null;
+  password?: string | null;
+  token?: string | null;
+  realm?: string | null;
+}
+
+/**
+ * Data format options for the Infinispan client.
+ * @public
+ */
+export interface InfinispanDataFormatOptions {
+  keyType?: string | null;
+  valueType?: string | null;
+  mediaType?: 'text/plain' | 'application/json' | null;
+}
+
+/**
+ * Detailed client behavior options for the Infinispan client.
+ * @public
+ */
+export interface InfinispanClientBehaviorOptions {
+  version?: '2.9' | '2.5' | '2.2' | null;
+  cacheName?: string | null;
+  maxRetries?: number | null;
+  connectionTimeout?: number | null;
+  socketTimeout?: number | null;
+  authentication?: InfinispanAuthOptions | null;
+  ssl?: InfinispanSslOptions | null;
+  dataFormat?: InfinispanDataFormatOptions | null;
+  topologyUpdates?: boolean | null;
```

Comment:
will people actually for real pass in nulls here? Won't they just leave out the key instead? Same in all of the types above, that seems weird